js/validator.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Created by peterwilkins on 02/12/2015.
+ */
+
+
+
+$(function () {
+
+    $('#target').submit(function (e) {
+        e.preventDefault();
+        // required
+        if (!$('#field1').val()) {
+            $("#err1").alert('close');
+            $('#field1').after(
+                '<div class="alert alert-danger alert-dismissable" id="err1">' +
+                '<button type="button" class="close" ' +
+                'data-dismiss="alert" aria-hidden="true">' +
+                '&times;' +
+                '</button>' +
+                'Required field' +
+                '</div>');
+            console.log('field1 is empty');
+        }
+        else {
+            $("#err1").alert('close');
+        }
+
+        //Max length 8
+        if ($('#field2').val().length >= 8) {
+            $("#err2").alert('close');
+            $('#field2').after(
+                '<div class="alert alert-danger alert-dismissable" id="err2">' +
+                '<button type="button" class="close" ' +
+                'data-dismiss="alert" aria-hidden="true">' +
+                '&times;' +
+                '</button>' +
+                'Max length 8' +
+                '</div>');
+            console.log('field2 longer than 8');
+        }
+        else {
+            $("#err2").alert('close');
+        }
+        //Required - Min length 10 - Max length 25
+        if ($('#field3').val().length < 10 || $('#field3').val().length > 25) {
+            $("#err3").alert('close');
+            $('#field3').after(
+                '<div class="alert alert-danger alert-dismissable" id="err3">' +
+                '<button type="button" class="close" ' +
+                'data-dismiss="alert" aria-hidden="true">' +
+                '&times;' +
+                '</button>' +
+                'Required field, Min length 10 - Max length 25' +
+                '</div>');
+            console.log('field3 wrong');
+        }
+        else {
+            $("#err3").alert('close');
+        }
+
+        //Not required - Min length 10 - Max length 25
+        if ($('#field4').val()) {
+            if ($('#field4').val().length < 10 || $('#field4').val().length > 25) {
+                $("#err4").alert('close');
+                $('#field4').after(
+                    '<div class="alert alert-danger alert-dismissable" id="err4">' +
+                    '<button type="button" class="close" ' +
+                    'data-dismiss="alert" aria-hidden="true">' +
+                    '&times;' +
+                    '</button>' +
+                    'Not Required field, Min length 10 - Max length 25' +
+                    '</div>');
+                console.log('field4 wrong');
+            }
+            else {
+                $("#err4").alert('close');
+            }
+        }
+
+        //Letters only
+        var testLetters = /^[a-zA-Z]+$/;
```

Comment:
this doesnt allow spaces. I no the validation is ambiguous about this but how would you make it allow spaces if you wanted to?